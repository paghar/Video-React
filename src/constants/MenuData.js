//@flow

export type MenuDataType = {
  id: number,
  menuIcon: string,
  menuTitle: string,
  menuLink: string,
};

export const MenuData = [
  { id: 1, menuIcon: '61449', menuTitle: 'Workspace', menuLink: '/home' },
  { id: 2, menuIcon: '61474', menuTitle: 'Projects', menuLink: '/home' },
  { id: 3, menuIcon: '61614', menuTitle: 'Board', menuLink: '/home' },
  { id: 4, menuIcon: '61633', menuTitle: 'Links', menuLink: '/home' },
];
