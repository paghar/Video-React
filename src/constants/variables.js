//@flow

export const MEDIA_BROWSER_RIGHT_SIDE_MENU_ANIMATION_DURATION = 300; // in ms
export const LEFT_MENU_ANIMATION_DURATION = 300; // in ms
export const FOLDER_TYPE = 'FOLDER_TYPE';
export const FILE_TYPE = 'FILE_TYPE';
export const POP_UP_LINE = 'HR';
export const POP_UP_HAS_SUBITEMS = 'SUBITEM';
