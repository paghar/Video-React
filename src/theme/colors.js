import theme from 'styled-theming';

export const Generals = {
  i: theme('mode', {
    dark: '#05050d',
  }),
  ii: theme('mode', {
    dark: '#0d0d1a',
  }),
  iii: theme('mode', {
    dark: '#151526',
  }),
  iv: theme('mode', {
    dark: '#212133',
  }),
  v: theme('mode', {
    dark: '#262640',
  }),
  vi: theme('mode', {
    dark: '#7b7b9a',
  }),
  vii: theme('mode', {
    dark: '#b7b7cc',
  }),
  viii: theme('mode', {
    dark: '#ffffff',
  }),
  x: theme('mode', {
    dark: '#e6e6f3',
  }),
  ix: theme('mode', {
    dark: '#f1f1f9',
  }),
};

export const Primary = {
  Dim: theme('mode', {
    dark: '#2172ff',
  }),
  Light: theme('mode', {
    dark: '#317BFC',
  }),
  Dark: theme('mode', {
    dark: '#0E5EE8',
  }),
};

export const PromptWarn = {
  Dim: theme('mode', {
    dark: '#F51653',
  }),
  Light: theme('mode', {
    dark: '#FF4074',
  }),
  Dark: theme('mode', {
    dark: '#DE0540',
  }),
};

export const Secondary = {
  i: theme('mode', {
    dark: '#00d3b2',
  }),
  ii: theme('mode', {
    dark: '#ffcc00',
  }),
  iii: theme('mode', {
    dark: '#f51653',
  }),
  iv: theme('mode', {
    dark: '#64a7ff',
  }),
};

export const Avatars = [
  theme('mode', {
    dark: '#5c4dff',
  }),
  theme('mode', {
    dark: '#685ce6',
  }),
  theme('mode', {
    dark: '#7d73e6',
  }),
  theme('mode', {
    dark: '#928ae6',
  }),
];

export const Gradient = theme('mode', {
  dark:
    'linear-gradient(to bottom, rgba(21, 21, 38, 0), rgba(21, 21, 38, 0.69) 53%, #111120)',
});
