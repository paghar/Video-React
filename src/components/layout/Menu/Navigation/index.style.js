//@flow
import styled from 'styled-components';
import { Generals, Primary } from 'theme/colors';
import { Wrapper } from 'components/shared/Typography/index.style';
import Avatar from 'components/shared/Avatar';
import Typography from 'components/shared/Typography';
import Icon from 'components/shared/Icon';

export const LogoContainer = styled.div`
  display: inline-flex;
  align-items: center;
  justify-content: center;
  cursor: pointer;
  margin: 21px 0px 50px 12px;
`;

export const Logo = styled(Avatar)`
  margin: 0px 12px 0px 0px;
  align-self: center;
`;

export const Text = styled(Typography)`
  align-self: center;
`;

export const IconItem = styled(Icon)`
  margin: 0px 16px 0px 0px;
  align-self: center;
`;
export const MenuItem = styled.div`
  padding: 10px 0px 10px 0px;
  display: flex;
  flex-direction: row;
  cursor: pointer;
  ::before{
    content: ' ';
    background-color: ${Primary.Dim};
    width: 4px; 
    height:30px
    margin: 0px 14px 0px 0px;
    visibility: hidden;
  }
  &:hover {
    ${IconItem} {
      color: ${Generals.viii};
    }
    ${Wrapper} {
      color: ${Generals.viii};
    }
    ::before{
      visibility: visible;
    }
  }
`;
