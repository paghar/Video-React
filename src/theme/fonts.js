import theme from 'styled-theming';

export const Title1 = {
  fontName: theme('size', {
    medium: "'Red Hat Text', sans-serif",
  }),
  fontSize: theme('size', {
    medium: '22px',
  }),
  fontWeight: theme('size', {
    medium: '400',
  }),
  lineHeight: theme('size', {
    medium: '35px',
  }),
};

export const Title2 = {
  fontName: theme('size', {
    medium: "'Red Hat Text', sans-serif",
  }),
  fontSize: theme('size', {
    medium: '21px',
  }),
  fontWeight: theme('size', {
    medium: '500',
  }),
  lineHeight: theme('size', {
    medium: '33px',
  }),
};

export const Display1 = {
  fontName: theme('size', {
    medium: "'Red Hat Text', sans-serif",
  }),
  fontSize: theme('size', {
    medium: '14px',
  }),
  fontWeight: theme('size', {
    medium: '400',
  }),
  lineHeight: theme('size', {
    medium: '23px',
  }),
};

export const Display1M = {
  fontName: theme('size', {
    medium: "'Red Hat Text', sans-serif",
  }),
  fontSize: theme('size', {
    medium: '14px',
  }),
  fontWeight: theme('size', {
    medium: '500',
  }),
  lineHeight: theme('size', {
    medium: '23px',
  }),
};

export const Display2 = {
  fontName: theme('size', {
    medium: "'Red Hat Text', sans-serif",
  }),
  fontSize: theme('size', {
    medium: '13px',
  }),
  fontWeight: theme('size', {
    medium: '400',
  }),
  lineHeight: theme('size', {
    medium: '21px',
  }),
};
export const Display3 = {
  fontName: theme('size', {
    medium: "'Red Hat Text', sans-serif",
  }),
  fontSize: theme('size', {
    medium: '13px',
  }),
  fontWeight: theme('size', {
    medium: '500',
  }),
  lineHeight: theme('size', {
    medium: '21px',
  }),
};
export const Display4 = {
  fontName: theme('size', {
    medium: "'Red Hat Text', sans-serif",
  }),
  fontSize: theme('size', {
    medium: '12px',
  }),
  fontWeight: theme('size', {
    medium: '400',
  }),
  lineHeight: theme('size', {
    medium: '20px',
  }),
};
export const Display5 = {
  fontName: theme('size', {
    medium: "'Red Hat Text', sans-serif",
  }),
  fontSize: theme('size', {
    medium: '11px',
  }),
  fontWeight: theme('size', {
    medium: '400',
  }),
  lineHeight: theme('size', {
    medium: '18px',
  }),
};
export const Display6 = {
  fontName: theme('size', {
    medium: "'Red Hat Text', sans-serif",
  }),
  fontSize: theme('size', {
    medium: '11px',
  }),
  fontWeight: theme('size', {
    medium: '500',
  }),
  lineHeight: theme('size', {
    medium: '18px',
  }),
};
