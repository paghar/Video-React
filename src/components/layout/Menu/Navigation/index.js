//@flow
import React from 'react';
import { Logo, Text, IconItem, MenuItem, LogoContainer } from './index.style';
import { Display1, Display3 } from 'theme/fonts';
import { Generals } from 'theme/colors';
import type { MenuDataType } from 'constants/MenuData';

type Props = {
  logoBackgroundColor: string,
  logoImgUrl?: string,
  logoText: string,
  menuData: Array<MenuDataType>,
  logoShortName: string,
};

function Navigation({
  logoBackgroundColor,
  logoImgUrl,
  logoText,
  menuData,
  logoShortName,
}: Props) {
  return (
    <div>
      <LogoContainer>
        <Logo
          backgroundColor={logoBackgroundColor}
          font={Display3}
          imgUrl={logoImgUrl}
          color={Generals.viii}
          width="31px"
          height="31px"
          borderRadius="100%"
        >
          {logoShortName}
        </Logo>

        <Text font={Display1} color={Generals.vii} hoverable>
          {logoText}
        </Text>
      </LogoContainer>

      {menuData.map(item => (
        <MenuItem key={item.id}>
          <IconItem
            fontSize="16px"
            color={Generals.vii}
            hoverColor={Generals.viii}
            iconNumber={item.menuIcon}
          ></IconItem>
          <Text font={Display1} hoverable color={Generals.vi}>
            {item.menuTitle}
          </Text>
        </MenuItem>
      ))}
    </div>
  );
}

export default Navigation;
