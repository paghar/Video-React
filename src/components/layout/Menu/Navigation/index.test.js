//@flow
import React from 'react';
import renderer from 'react-test-renderer';
import Navigation from '.';
import { MenuData } from 'constants/MenuData';
import logoImage from 'assets/image/avasam-1.png';
import { Avatars } from 'theme/colors';

describe('<Navigation />', () => {
  it('when menu in open mode', () => {
    const tree = renderer
      .create(
        <Navigation
          logoBackgroundColor="#fff"
          logoImgUrl={logoImage}
          logoText="Avasam Agency"
          logoShortName="AA"
          menuData={MenuData}
        ></Navigation>
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
});

describe('<Navigation />', () => {
  it('when menu in close mode', () => {
    const tree = renderer
      .create(
        <Navigation
          isShrinked
          logoBackgroundColor="#fff"
          logoImgUrl={logoImage}
          logoText="Avasam Agency"
          logoShortName="AA"
          menuData={MenuData}
        ></Navigation>
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
});
