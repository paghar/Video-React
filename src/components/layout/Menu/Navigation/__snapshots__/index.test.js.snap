// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Navigation /> when menu in close mode 1`] = `
.c3 {
  pointer-events: none;
  -moz-user-select: none;
  -khtml-user-select: none;
  -webkit-user-select: none;
  -ms-user-select: none;
  -webkit-user-select: none;
  -moz-user-select: none;
  -ms-user-select: none;
  user-select: none;
}

.c1 {
  background-color: #fff;
  background-image: url(avasam-1.png);
  background-size: 100% 100%;
  width: 31px;
  height: 31px;
  border-radius: 100%;
  border: 0px solid #0d0d1a;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  outline: none;
}

.c6 {
  font-family: 'FontAwesome' !important;
  font-size: 16px;
  margin: 0px;
  padding: 0px;
  -webkit-align-self: center;
  -ms-flex-item-align: center;
  align-self: center;
  -moz-user-select: none;
  -khtml-user-select: none;
  -webkit-user-select: none;
  -ms-user-select: none;
  -webkit-user-select: none;
  -moz-user-select: none;
  -ms-user-select: none;
  user-select: none;
}

.c0 {
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  cursor: pointer;
  margin: 21px 0px 50px 12px;
}

.c2 {
  margin: 0px 12px 0px 0px;
  -webkit-align-self: center;
  -ms-flex-item-align: center;
  align-self: center;
}

.c4 {
  -webkit-align-self: center;
  -ms-flex-item-align: center;
  align-self: center;
}

.c7 {
  margin: 0px 16px 0px 0px;
  -webkit-align-self: center;
  -ms-flex-item-align: center;
  align-self: center;
}

.c5 {
  padding: 10px 0px 10px 0px;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: row;
  -ms-flex-direction: row;
  flex-direction: row;
  cursor: pointer;
}

.c5::before {
  content: ' ';
  width: 4px;
  height: 30px;
  margin: 0px 14px 0px 0px;
  visibility: hidden;
}

.c5:hover::before {
  visibility: visible;
}

<div>
  <div
    className="c0"
  >
    <button
      className="c1 c2"
      height="31px"
      width="31px"
    />
    <span
      className="c3 c4"
    >
      Avasam Agency
    </span>
  </div>
  <div
    className="c5"
  >
    <span
      className="c6 c7"
      fontSize="16px"
    >
      
    </span>
    <span
      className="c3 c4"
    >
      Workspace
    </span>
  </div>
  <div
    className="c5"
  >
    <span
      className="c6 c7"
      fontSize="16px"
    >
      
    </span>
    <span
      className="c3 c4"
    >
      Projects
    </span>
  </div>
  <div
    className="c5"
  >
    <span
      className="c6 c7"
      fontSize="16px"
    >
      
    </span>
    <span
      className="c3 c4"
    >
      Board
    </span>
  </div>
  <div
    className="c5"
  >
    <span
      className="c6 c7"
      fontSize="16px"
    >
      
    </span>
    <span
      className="c3 c4"
    >
      Links
    </span>
  </div>
</div>
`;

exports[`<Navigation /> when menu in open mode 1`] = `
.c3 {
  pointer-events: none;
  -moz-user-select: none;
  -khtml-user-select: none;
  -webkit-user-select: none;
  -ms-user-select: none;
  -webkit-user-select: none;
  -moz-user-select: none;
  -ms-user-select: none;
  user-select: none;
}

.c1 {
  background-color: #fff;
  background-image: url(avasam-1.png);
  background-size: 100% 100%;
  width: 31px;
  height: 31px;
  border-radius: 100%;
  border: 0px solid #0d0d1a;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  outline: none;
}

.c6 {
  font-family: 'FontAwesome' !important;
  font-size: 16px;
  margin: 0px;
  padding: 0px;
  -webkit-align-self: center;
  -ms-flex-item-align: center;
  align-self: center;
  -moz-user-select: none;
  -khtml-user-select: none;
  -webkit-user-select: none;
  -ms-user-select: none;
  -webkit-user-select: none;
  -moz-user-select: none;
  -ms-user-select: none;
  user-select: none;
}

.c0 {
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  cursor: pointer;
  margin: 21px 0px 50px 12px;
}

.c2 {
  margin: 0px 12px 0px 0px;
  -webkit-align-self: center;
  -ms-flex-item-align: center;
  align-self: center;
}

.c4 {
  -webkit-align-self: center;
  -ms-flex-item-align: center;
  align-self: center;
}

.c7 {
  margin: 0px 16px 0px 0px;
  -webkit-align-self: center;
  -ms-flex-item-align: center;
  align-self: center;
}

.c5 {
  padding: 10px 0px 10px 0px;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: row;
  -ms-flex-direction: row;
  flex-direction: row;
  cursor: pointer;
}

.c5::before {
  content: ' ';
  width: 4px;
  height: 30px;
  margin: 0px 14px 0px 0px;
  visibility: hidden;
}

.c5:hover::before {
  visibility: visible;
}

<div>
  <div
    className="c0"
  >
    <button
      className="c1 c2"
      height="31px"
      width="31px"
    />
    <span
      className="c3 c4"
    >
      Avasam Agency
    </span>
  </div>
  <div
    className="c5"
  >
    <span
      className="c6 c7"
      fontSize="16px"
    >
      
    </span>
    <span
      className="c3 c4"
    >
      Workspace
    </span>
  </div>
  <div
    className="c5"
  >
    <span
      className="c6 c7"
      fontSize="16px"
    >
      
    </span>
    <span
      className="c3 c4"
    >
      Projects
    </span>
  </div>
  <div
    className="c5"
  >
    <span
      className="c6 c7"
      fontSize="16px"
    >
      
    </span>
    <span
      className="c3 c4"
    >
      Board
    </span>
  </div>
  <div
    className="c5"
  >
    <span
      className="c6 c7"
      fontSize="16px"
    >
      
    </span>
    <span
      className="c3 c4"
    >
      Links
    </span>
  </div>
</div>
`;
